src/fromager/sources.py

```diff
@@ -281,6 +284,18 @@ def patch_source(ctx: context.WorkContext, source_root_dir: pathlib.Path) -> Non
             )
 
 
+def _warn_for_old_patch(source_root_dir: pathlib.Path, version) -> None:
+    # Get the existing version and req name as per the source_root_dir naming conventions
+    req_name = source_root_dir.name.split("-")[0]
+    existing_version = Version(source_root_dir.name.split("-")[-1])
```

Comment:
Something that I learned from @tiran 
```suggestion
    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
```